### PR TITLE
V704-034: Limit number of logs in $ALS_HOME/.als

### PR DESCRIPTION
**Pull request by AdrienBoulanger**
Add logThreshold property to limit the number of logs.

### CODE REVIEW

File: source/ada/lsp-ada_handlers.adb [hunk 1]
```diff
@@ -531,6 +531,30 @@ package body LSP.Ada_Handlers is
...
+   procedure Clean_Logs (Self : access Message_Handler; Dir : Virtual_File) is
+      Log_Files : File_Array_Access := Read_Dir (Dir, Files_Only);
+      Dummy     : Boolean;
+      Cpt       : Integer := 0;
+   begin
+      Sort (Log_Files.all);
+      --  Browse the log files in reverse timestamp order
+      for F of reverse Log_Files.all loop
+         --  Filter out files like traces.cfg
+         if GNATCOLL.Utils.Starts_With (+F.Base_Name, "als.") then
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_handlers.adb:547]: You should filter with .txt instead: the log filenames may be changed in traces.cfg

File: source/ada/lsp-ada_handlers.adb [hunk 2]
```diff
@@ -531,6 +531,30 @@ package body LSP.Ada_Handlers is
       LSP.File_Monitors.Unchecked_Free (Self.File_Monitor);
    end Cleanup;
 
+   ----------------
+   -- Clean_Logs --
+   ----------------
+
+   procedure Clean_Logs (Self : access Message_Handler; Dir : Virtual_File) is
+      Log_Files : File_Array_Access := Read_Dir (Dir, Files_Only);
```
Comment (AnthonyLeonardoGracio) [inline source/ada/lsp-ada_handlers.adb:539]: "Files" might be a better name, since it may not only contain log files

**Review by AnthonyLeonardoGracio — CHANGES_REQUESTED** [review]

File: source/ada/lsp-ada_handlers.adb (hunk 1 above)
Comment (AdrienBoulanger) [reply source/ada/lsp-ada_handlers.adb]: The default name is finishing by ".log" and we are overriding the name in traces.cfg when using ALS_HOME

File: source/ada/lsp-ada_handlers.adb (hunk 2 above)
Comment (AdrienBoulanger) [reply source/ada/lsp-ada_handlers.adb]: Done

File: source/ada/lsp-ada_handlers.adb (hunk 1 above)
Comment (AnthonyLeonardoGracio) [reply source/ada/lsp-ada_handlers.adb]: Then check for '.log' no ? Also, will it conflict with the GS preference?

**Review by AnthonyLeonardoGracio — APPROVED** [review]

**Pull request merged**